cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/cache/rate_limiting_queue.go

```diff
@@ -317,6 +322,30 @@ func (q *Type) isProcessing(item string) bool {
 	return q.processing.Has(item)
 }
 
+// Atomic check if not already processing then Add.
```

Comment:
The docstring sounds nice and simple, and I can map the `Atomic` to the lock and unlock (great), the `if not already processing` to a single line, but the `then Add` appears to do a bunch of things that I don't understand without context (eg. why `expected.Delete()` or why an early return if `q.dirty.Has()` etc.). Maybe a few more comments inside that conditional would help people who read this in the future?

Reply:
two things:
1) the explanation for` expected.Delete() ` or the whole reason behind `expected sets.String` in the queue is to avoid a race condition in unit tests, where an item is added to the queue and then the test code needs to [wait until its been processed](https://github.com/vmware-tanzu/kubeapps/blob/fe00fbb366b78ae21b413b944e44b76a6f7d69e5/cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/cache/rate_limiting_queue.go#L321) by the queue before taking further action. 
2) q.dirty, etc comes from the original [queue](https://github.com/kubernetes/client-go/blob/62756eea6f0e65cf1bda27ac7f96e754f2f90951/util/workqueue/queue.go#L78) which was the inspiration for this code